### PR TITLE
Fix headings indent

### LINKED ISSUE #91200: The first heading in all doc comments have left margin
Which looks weird as you can see:

Screenshot from 2021-11-24 22-21-15

It's even more weird on enum variants:

Screenshot from 2021-11-24 22-24-04

I'll fix it tomorrow.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #91200.

Screenshots with the fix:

Screenshot from 2021-11-25 15-32-35
Screenshot from 2021-11-25 15-32-49

If the first element of a top docblock is a heading, we still need to keep the indent, but only on this one (I added a test to check it). We need it because otherwise the anchor will go over the `[-]` toggle.

cc @camelid 
r? @jsha

### CODE REVIEW

**Review by jsha — COMMENTED** [review]
Nice, thanks for fixing this - it had been bugging me too and I never got around to filing an issue. :-)

@bors r+ rollup

**Review by jsha — APPROVED** [review]

**Comment by camelid** [comment]
> The first commit comes from #91181 to make things easier.

Wait, we shouldn't be landing both of these PRs at the same time. I'm unapproving for now, please re-approve once the other PR has landed and you've rebased over master.

@bors r-

**Comment by camelid** [comment]
Also, @jsha looks like bors ignores commands in reviews?

**Comment by camelid** [comment]
> The first commit comes from #91181 to make things easier.

And please remove this line once you've rebased too.

**Comment by GuillaumeGomez** [comment]
Rebased on master so #91181 is now included and removed from the commit list. :)

@bors: r=jsha rollup

**Pull request merged**
